src/v/cluster/health_monitor_types.h

```diff
@@ -239,9 +242,73 @@ struct cluster_health_report
     operator==(const cluster_health_report&, const cluster_health_report&)
       = default;
 
-    auto serde_fields() {
-        return std::tie(
-          raft0_leader, node_states, node_reports, bytes_in_cloud_storage);
+    cluster_health_report copy() const;
+
+    ss::future<> serde_async_write(iobuf& out) {
+        using serde::write;
+        using serde::write_async;
+        // the current version decodes into the decoded version and is used in
+        // request handling--that is, it is used at layer above serialization so
+        // without further changes we'll need to preserve that behavior.
+        write(out, raft0_leader);
+        write(out, node_states);
+        write(out, static_cast<serde::serde_size_t>(node_reports.size()));
+        for (auto& nr : node_reports) {
+            co_await write_async(out, *nr);
+        }
+        write(out, bytes_in_cloud_storage);
+    }
+
+    ss::future<> serde_async_read(iobuf_parser& in, const serde::header& h) {
```

Comment:
there is a lot going on in this one commit. is it all related?

Reply:
yes, i know it is a lot of changes for one commit, it is indeed related as this one change (a type that we use to propagate `cluster_health_reports`) is influencing a lot of other places like: 
- serialization
- callers of health_monitor_frontend apis
- tests